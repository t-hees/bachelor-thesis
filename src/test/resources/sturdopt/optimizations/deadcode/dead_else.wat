(module
  (func $add (param $lhs i32) (param $rhs i32) (result i32)
    i32.const 1
    (if (result i32)
      (then
        local.get $lhs
        local.get $rhs
        i32.add
      )
      (else
        i32.const 0
        (if (result i32)
          (then
            i32.const 1
          )
          (else
            i32.const 0
          )
        )
      )
    )
  )
  (func (export "main") (result i32)
    i32.const 2
    i32.const 3
    call $add)
)
