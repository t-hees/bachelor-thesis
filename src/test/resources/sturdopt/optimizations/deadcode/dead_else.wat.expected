(module
  (func $add (param $lhs i32) (param $rhs i32) (result i32)
    i32.const 1
    drop
    local.get $lhs
    local.get $rhs
    i32.add
  )
  (func (export "main") (result i32)
    i32.const 2
    i32.const 3
    call $add)
)
